### ISSUE TITLE
Change default value for KeyExpirationEventMessageListener to disable notifications

### ISSUE THREAD

**Issue by jxblum** [spring-projects/spring-data-redis#2670]
The [default value](https://github.com/spring-projects/spring-data-redis/blob/3.2.0-M1/src/main/java/org/springframework/data/redis/listener/KeyspaceEventMessageListener.java#L43) in Spring Data Redis for `notify-keyspace-events` is set to `EA`. Therefore, when a user declares and registers a bean of type `KeyExpirationEventMessageListener` (which [extends](https://github.com/spring-projects/spring-data-redis/blob/3.2.0-M1/src/main/java/org/springframework/data/redis/listener/KeyExpirationEventMessageListener.java#L32) `KeyspaceEventMessageListener`) in the Spring `ApplicationContext`, Spring Data Redis will [send](https://github.com/spring-projects/spring-data-redis/blob/3.2.0-M1/src/main/java/org/springframework/data/redis/listener/KeyspaceEventMessageListener.java#L88) a `config set notify-keyspace-events EA` command to the Redis server on [initialization](https://github.com/spring-projects/spring-data-redis/blob/3.2.0-M1/src/main/java/org/springframework/data/redis/listener/KeyspaceEventMessageListener.java#L123-L126) (and then [init()](https://github.com/spring-projects/spring-data-redis/blob/3.2.0-M1/src/main/java/org/springframework/data/redis/listener/KeyspaceEventMessageListener.java#L73-L97)).

While this may be "convenient" during development, it is NOT (ever) appropriate for production deployments, and can lead to confusing behavior exactly like [this](https://github.com/spring-projects/spring-data-redis/issues/2654).

With a combination of _Lettuce_ attempting to reconnect (_Jedis_ is unable to reconnect) and the fact that Redis does not retain configuration changes between restarts (i.e. non-persistent config when changed at runtime; even when applied using the `redis-cli`), then this situation, such as in Issue #2654, can occur.

> NOTE: It is not uncommon for data stores to retain (persist) dynamic configuration changes between sessions.

Therefore, I recommend the default value in Spring Data Redis be changed to empty, precisely how the Redis server (in `redis.conf` from a clean install) defines this setting, itself. 

If users change the default (e.g. to `EA`, or otherwise), then they have chosen to "opt-in" and are therefore responsible for the change, such as when the Redis server needs to be restarted. 

As explained in Issue #2654, the appropriate location to declare "infrastructure-related" configuration settings to the Redis server is in the `redis.conf` file, where it belongs! Additionally, users must ensure any custom (modified) configuration file is then passed to and recognized by the Redis server on start.

> TIP: On a positive note, the customized/modified `redis.conf` used to change the default behavior of the Redis server can be checked into version control the same as the application source code that requires the Redis server to behave in a custom way.

In conclusion, the framework, or any library for that matter, should never be making "infrastructure" decisions for users.
